jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/json/JsonCfnParser.kt

```diff
@@ -0,0 +1,395 @@
+
+        val topLevelProperties = mutableMapOf<String, CfnNamedNode>()
+
+        for (property in obj.propertyList) {
+            val propertyName = property.name
+
+            if (!TOP_LEVEL_RESOURCE_PROPERTIES.contains(propertyName)) {
+//                addProblemOnNameElement(property, message("format.unknown.resource.property", propertyName))
+            }
+
+            val node = when (propertyName) {
+                DEPENDS_ON -> handleResourceDependsOn(property)
+                TYPE -> handleResourceType(property)
+                CONDITION -> handleResourceCondition(property)
+                PROPERTIES -> handleResourceProperties(property)
+                else -> {
+                    CfnNameValueNode(keyName(property), property.value?.let { handleExpression(it, true) }).registerNode(property)
+                }
+            }
+
+            topLevelProperties[propertyName] = node
+        }
+
+        val properties = topLevelProperties.values
+        return CfnResourceNode(
+            key,
+            properties.findSection<CfnResourceTypeNode>(),
+            properties.findSection<CfnResourcePropertiesNode>(),
+            properties.findSection<CfnResourceConditionNode>(),
+            properties.findSection<CfnResourceDependsOnNode>(),
+            topLevelProperties
+        ).registerNode(resourceProperty)
+    }
+
+    private fun handleResourceDependsOn(property: JsonProperty): CfnResourceDependsOnNode {
+        val values = checkAndGetStringOrStringArray(property)
+        return CfnResourceDependsOnNode(keyName(property), values).registerNode(property)
+    }
+
+    private fun handleResourceType(typeProperty: JsonProperty): CfnResourceTypeNode = CfnResourceTypeNode(
+        keyName(typeProperty),
+        checkAndGetStringElement(typeProperty.value)
+    ).registerNode(typeProperty)
+
+    private fun handleResourceCondition(property: JsonProperty): CfnResourceConditionNode =
+        CfnResourceConditionNode(keyName(property), checkAndGetStringElement(property.value)).registerNode(property)
+
+    private fun handleResourceProperties(propertiesProperty: JsonProperty): CfnResourcePropertiesNode {
+        val value = checkAndGetObject(propertiesProperty.value)
+            ?: return CfnResourcePropertiesNode(keyName(propertiesProperty), emptyList()).registerNode(propertiesProperty)
+
+        val propertyNodes = value.propertyList.mapNotNull { property ->
+            CfnResourcePropertyNode(keyName(property), property.value?.let { handleExpression(it, allowFunctions = true) }).registerNode(property)
+        }
+
+        return CfnResourcePropertiesNode(keyName(propertiesProperty), propertyNodes).registerNode(propertiesProperty)
+    }
+
+    private fun handleConditions(conditions: JsonProperty): CfnConditionsNode = parseNameValues(
+        conditions,
+        { node -> CfnConditionNode(keyName(node), handleExpression(node.value!!, allowFunctions = true)).registerNode(node) },
```

Comment:
It seems like this would make more sense to not have the `!!` and get handled by handleExpression if it is null, ditto for the other case of this

Reply:
it has been removed.

No more `!!`!